tests/test_rest_handler.py

```diff
@@ -210,55 +264,134 @@ def test_rest_handler_check_token_true(mock_rest_sms):
+})
 def test_rest_handler_check_token_false(mock_rest_sms):
     
     obj = TokenAuth()
     obj.b_token = 'different_token'
     response = obj.checkToken()
 
     assert response == False
+
+@patch.object(TokenAuth, 'secretManagerGet', return_value = 'bearer_token')
+@patch.dict(os.environ, {
+    "HAAS_AUTH_USERNAME_KEY":"key", 
+    "HAAS_AUTH_PASSWORD_KEY":"key",
+    "HAAS_BEARER_TOKEN_KEY": "key", 
+    "HAAS_REFRESH_TOKEN_KEY": "key",
+    "HAAS_AUTH_ENDPOINT": "auth_endpoint",
+    "HAAS_API_ENDPOINT": "api_endpoint",
+    "PROJECT_ID":"random_id"
+})
+def test_rest_handler_token_update(mock_rest_sms):
+    
+    obj = TokenAuth()
+    response = obj.tokenUpdate()
+
+    assert response == 'bearer_token'
+
+@patch.object(TokenAuth, 'secretManagerGet', return_value = 'bearer_token')
+@patch.dict(os.environ, {
+    "HAAS_AUTH_USERNAME_KEY":"key", 
+    "HAAS_AUTH_PASSWORD_KEY":"key",
+    "HAAS_BEARER_TOKEN_KEY": "key", 
+    "HAAS_REFRESH_TOKEN_KEY": "key",
+    "HAAS_AUTH_ENDPOINT": "auth_endpoint",
+    "HAAS_API_ENDPOINT": "api_endpoint",
+    "PROJECT_ID":"random_id"
+})
+def test_rest_handler_get_token(mock_rest_sms):
     
-@patch.object(TokenAuth, 'secretManagerGet', return_value = 'password')
+    obj = TokenAuth()
+    obj.b_token = 'bearer_token'
+    response = obj.getToken()
+
+    assert response == 'bearer_token'
+
+
+
+
+@patch.object(TokenAuth, 'secretManagerGet', return_value = 'bearer_token')
+@patch.object(TokenAuth, 'secretManagerSet')
 @patch.dict(os.environ,{"HAAS_AUTH_USERNAME_KEY":"key", "HAAS_AUTH_PASSWORD_KEY":"key", 
 "HAAS_BEARER_TOKEN_KEY": "key", "HAAS_REFRESH_TOKEN_KEY": "key", "HAAS_API_ENDPOINT":"endpoint"},clear=True)
-def test_rest_handler_check_password_true(mock_rest_sms):
-    
+def test_rest_handler_getOrganizations(mock_rest_smg, mock_rest_sms):
+    with open('tests/support/sample_organizations.json', 'r') as f:
+        organizations_json = json.load(f)
+
+    haas_response = requests.Response
+    haas_response.status_code = 200
+    haas_response.content = json.dumps(organizations_json)
+    requests.get = MagicMock(return_value = haas_response)
+
     obj = TokenAuth()
-    obj.api_password = 'password'
-    response = obj.checkPassword()
+    obj.b_token= 'bearer_token'
+    response = obj.getOrganizations()
 
-    assert response == True
-    
-@patch.object(TokenAuth, 'secretManagerGet', return_value = 'password')
+    assert response == organizations_json
+
+@patch.object(TokenAuth, 'secretManagerGet', return_value = None)
+@patch.object(TokenAuth, 'signIn', return_value = "bearer_token")
+@patch.object(TokenAuth, 'secretManagerSet')
 @patch.dict(os.environ,{"HAAS_AUTH_USERNAME_KEY":"key", "HAAS_AUTH_PASSWORD_KEY":"key", 
 "HAAS_BEARER_TOKEN_KEY": "key", "HAAS_REFRESH_TOKEN_KEY": "key", "HAAS_API_ENDPOINT":"endpoint"},clear=True)
-def test_rest_handler_check_password_false(mock_rest_sms):
-    
+def test_rest_handler_getOrganizations_fail(mock_rest_smg, mock_signIn, mock_rest_sms):
```

Comment:
This seems to test the same case as success above?

Reply:
Good catch, forgot to check the failed condition.